nixpkgs/nixpkgs.bzl

```diff
@@ -23,6 +26,38 @@ nixpkgs_git_repository = repository_rule(
   local = False,
 )
 
+
+def _nixpkgs_local_repository_impl(repository_ctx):
+  repository_ctx.file('BUILD')
+  if not \
+     bool(repository_ctx.attr.nix_file) != \
+     bool(repository_ctx.attr.nix_file_content):
+    fail("Specify one of 'nix_file' or 'nix_file_content' (but not both).")
+  if repository_ctx.attr.nix_file_content:
+    repository_ctx.file(
+      path = "default.nix",
+      content = repository_ctx.attr.nix_file_content,
+      executable = False,
+    )
+    target = repository_ctx.path("default.nix")
+  else:
+    target = repository_ctx.path(repository_ctx.attr.nix_file)
+    repository_ctx.symlink(target, target.basename)
+  repository_ctx.symlink(target.basename, repository_ctx.name)
```

Comment:
This is the symlink that produces the file for the `@foo//:foo` target trick, right? I’d add an explanatory comment.

Reply:
OK. I will duplicate the comment from the function just above.